src/Storages/ExternalStream/Kafka/WriteBufferFromKafka.cpp

```diff
@@ -20,15 +21,21 @@ void WriteBufferFromKafka::nextImpl()
     if (!offset())
         return;
 
+    /// `write_to_topic` should have been called before the buffer starts to work.
+    assert(topic);
+
+    auto * part_id = new Int32(partition_id);
+
     auto err = rd_kafka_produce(
-        topic.get(),
+        topic,
+        /// we want to trigger the partitioner function, check KafkaSink.cpp
         RD_KAFKA_PARTITION_UA,
         RD_KAFKA_MSG_F_COPY | RD_KAFKA_MSG_F_BLOCK,
         working_buffer.begin(),
         offset(),
-        nullptr /* key */,
-        0 /* keylen */,
-        static_cast<void *>(this) /* opaque */);
+        "unused" /* key, even though we don't use the key, but it's needed to trigger the partitioner_cb */,
```

Comment:
Can we just use empty string like "" and keylen = 0 ? If no, use one byte string.

Reply:
Yes, empty string actually works, updated.